src/org/jetbrains/plugins/scala/lang/refactoring/move/ScalaMoveUtil.scala

```diff
@@ -44,6 +44,8 @@ object ScalaMoveUtil {
 
   def doMoveClass(@NotNull aClass: PsiClass, @NotNull moveDestination: PsiDirectory, withCompanion: Boolean): PsiClass = {
     var fileWasDeleted: Boolean = false
+    val maybeCompanion = if(withCompanion) ScalaPsiUtil.getBaseCompanionModule(aClass) else None
```

Comment:
need to extract this here before anything has actually been moved so we can use it once the class is moved